### PR TITLE
Save a reference to created async tasks, to avoid tasks potentially disappearing

### PR DESCRIPTION
**Pull request by anio**
> Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks. A task that isn’t referenced elsewhere may get garbage collected at any time, even before it’s done.
>
> [python docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task)

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `$ tox` pass with this change (including linting)?
- [ ] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

Please review the link: https://textual.textualize.io/blog/2023/02/11/the-heisenbug-lurking-in-your-async-code/

### CODE REVIEW

File: redis/asyncio/cluster.py [hunk 1]
```diff
@@ -1117,13 +1117,13 @@ def set_nodes(
         if remove_old:
             for name in list(old.keys()):
                 if name not in new:
-                    asyncio.create_task(old.pop(name).disconnect())
+                    task = asyncio.create_task(old.pop(name).disconnect())
```
Comment (dvora-h) [inline redis/asyncio/cluster.py:1120]:
```suggestion
                    task = asyncio.create_task(old.pop(name).disconnect())  # noqa
```

File: redis/asyncio/cluster.py [hunk 2]
```diff
@@ -1117,13 +1117,13 @@ def set_nodes(
...
 
         for name, node in new.items():
             if name in old:
                 if old[name] is node:
                     continue
-                asyncio.create_task(old[name].disconnect())
+                task = asyncio.create_task(old[name].disconnect())
```
Comment (dvora-h) [inline redis/asyncio/cluster.py:1126]:
```suggestion
                task = asyncio.create_task(old[name].disconnect())  # noqa
```

File: redis/asyncio/cluster.py (hunk 1 above)
Comment (dvora-h) [reply redis/asyncio/cluster.py:1120]: Skip linter check

File: redis/asyncio/cluster.py (hunk 2 above)
Comment (dvora-h) [reply redis/asyncio/cluster.py:1126]: same

**Review by dvora-h — CHANGES_REQUESTED** [review]

**Review by dvora-h — APPROVED** [review]

**Comment by jakob-keller** [comment]
@anio: Thanks for bringing the issue up!

I am not convinced that the fix is appropriate, though. As the [source you provide](https://textual.textualize.io/blog/2023/02/11/the-heisenbug-lurking-in-your-async-code/) mentions:

> The solution recommended in the docs is to keep a reference to the task for as long as you need it to live.

The way I understand it is, that we need to save the task reference beyond the scope of the local / loop variable. As it stands, the reference will be lost almost immediately, rendering the fix ineffective.

**Pull request merged**